File: src/main/java/engineer/carrot/warren/thump/command/minecraft/handler/ICommandHandler.java
package engineer.carrot.warren.thump.command.minecraft.handler;

import net.minecraft.command.ICommandSender;

import java.util.List;

public interface ICommandHandler {
    String getCommand();

    void processParameters(ICommandSender sender, String[] parameters);

    String getUsage();

    List<String> addTabCompletionOptions(ICommandSender sender, String[] parameters);
}

File: src/main/kotlin/engineer/carrot/warren/thump/command/minecraft/handler/ICommandHandler.kt
package engineer.carrot.warren.thump.command.minecraft.handler

import net.minecraft.command.ICommandSender

interface ICommandHandler {
    val command: String

    fun processParameters(sender: ICommandSender, parameters: Array<String>)

    val usage: String

    fun addTabCompletionOptions(sender: ICommandSender, parameters: Array<String>): List<String>
}
